pkg/server/cstorvolumeconfig/restore_endpoint.go

```diff
@@ -0,0 +1,495 @@
+	maxRetries := 5
+	for i := 0; i < maxRetries; i++ {
+		cvcObj, err := clientset.CstorV1().CStorVolumeConfigs(namespace).Get(cvcName, metav1.GetOptions{})
+		// If CVC is not found then wait for it to exist in etcd
+		if err != nil && !k8serror.IsNotFound(err) {
+			return err
+		}
+		if cvcObj.Status.Phase == cstor.CStorVolumeConfigPhaseBound {
+			// Which means all the CStorVolume related resources are created
+			return nil
+		}
+		klog.Errorf("waiting for CVC: %s in namespace: %s to become Bounded error: %v", cvcObj.Name, cvcObj.Namespace, err)
+		time.Sleep(2 * time.Second)
+	}
+	return errors.Errorf("CVC %s in namespace %s is not in Bound status", cvcName, namespace)
+}
+
+// restoreCreateValidationRequest will validate the restore creation request and
+// returns error if any validation failed
+func restoreCreateValidationRequest(restore *openebsapis.CStorRestore) error {
+	// namespace is expected
+	if !restore.Spec.Local && len(strings.TrimSpace(restore.Namespace)) == 0 {
+		return errors.Errorf("failed to create restore '%v': missing namespace", restore.Name)
+	}
+
+	// restore name is expected
+	if len(strings.TrimSpace(restore.Spec.RestoreName)) == 0 {
+		return errors.Errorf("failed to create restore: missing restore name")
+	}
+
+	// volume name is expected
+	if len(strings.TrimSpace(restore.Spec.VolumeName)) == 0 {
+		return errors.Errorf("failed to create restore '%v': missing volume name", restore.Name)
+	}
+
+	// restoreIP is expected
+	if len(strings.TrimSpace(restore.Spec.RestoreSrc)) == 0 {
+		return errors.Errorf("failed to create restore '%v': missing restoreSrc", restore.Name)
+	}
+
+	// storageClass is expected if restore is for local snapshot
+	if restore.Spec.Local && len(strings.TrimSpace(restore.Spec.StorageClass)) == 0 {
+		return errors.Errorf("failed to create restore '%v': missing storageClass", restore.Name)
+	}
+
+	// size is expected if restore is for local snapshot
+	if restore.Spec.Local && len(strings.TrimSpace(restore.Spec.Size.String())) == 0 {
+		return errors.Errorf("failed to create restore '%v': missing size", restore.Name)
+	}
+	return nil
+}
+
+func validateStorageClassParameters(scObj *storagev1.StorageClass) error {
+	if _, ok := scObj.Parameters["cstorPoolCluster"]; !ok {
+		return errors.Errorf("storageclass %s doesn't have cstorPoolCluster details", scObj.Name)
+	}
+	if _, ok := scObj.Parameters["replicaCount"]; !ok {
+		return errors.Errorf("storageclass %s doesn't have replica count", scObj.Name)
+	}
+	if _, err := strconv.Atoi(scObj.Parameters["replicaCount"]); err != nil {
+		return errors.Wrapf(err, "failed to convert replica count %s into integer", scObj.Parameters["replicaCount"])
+	}
+	return nil
+}
+
+// getNodeID returns the node name either from node topology only topology is
+// specified in the storageclass or from pool manager nodes
+func (rOps *restoreAPIOps) getNodeID(scObj *storagev1.StorageClass) (string, error) {
+	var labelKey, labelValue string
```

Comment:
now we don't need this change anymore ( NodeID)

Reply:
Hmm..